Rebase against the upstream + Use Node.js v20 to prepare che-code assembly

RomanNikitenko opened the pull request:
Some adaptation is required for downstream, so please do not merge it.

### What does this PR do?
Rebase against the upstream + Use Node.js v20 to prepare che-code assembly 

### What issues does this PR fix?
<!-- Please include any related issue from the Eclipse Che repository (or from another issue tracker). -->
https://github.com/eclipse-che/che/issues/22965
https://github.com/eclipse-che/che/issues/22963

### How to test this PR?
- PR jobs should be green
- testing of any use cases is useful as we pick up a lot of commits from the upstream


### Does this PR contain changes that override default upstream Code-OSS behavior?
- [ ] the PR contains changes in the [code](https://github.com/che-incubator/che-code/tree/main/code) folder (you can skip it if your changes are placed in a che extension )
- [ ] the corresponding items were added to the [CHANGELOG.md](https://github.com/che-incubator/che-code/blob/main/.rebase/CHANGELOG.md) file
- [ ] rules for automatic `git rebase` were added to the [.rebase](https://github.com/che-incubator/che-code/tree/main/.rebase) folder

azatsarynnyy reviewed (approved):
According to [these benchmarks](https://github.com/zertosh/v8-compile-cache?tab=readme-ov-file#benchmarks), disabling the compile cache shouldn't have a significant affect.
Thanks @RomanNikitenko for investigating the problem and providing the patch 👍

sandy081 committed:
fix #212674 (#213583)

hediet committed:
Fixes diff editor height in monaco editor playground example. (#213585)

pouyakary committed:
Feat: Bolder Typeface + Configurable Letter Spacing for Minimap's Section Header Labels ✨ (#209990)

* Rendering Section Header Labels with space in
between the characters for better readability.

* Making Section Headers Bold

* Added settings for the Section Header Letter
Spacing of the Minimap

* a more performant and non-breaking way to render characters

* fixed the git problems

* add test default value

* Allow fine-tuning the letter spacing, minor tweaks

---------

Co-authored-by: Alexandru Dima <alexdima@microsoft.com>

DonJayamanne committed:
Support diff merge editor for Notebooks (#213602)

Yoyokrazy committed:
Add Search in Cell Selection to Notebook Find Widget (#213409)

* all search scope functionality except cell decorations

* fix filter icon showing fake useage

* decorations + css tweaking + PR feedback

aiday-mar committed:
Adding accessibility help for verbose hover (#212783)

* adding code in order to provide accessibility help for the hover

* adding the description of the possible commands that can be used

* reusing the method

* joining the content and changing the text in the help view

* polishing the code

* removing the question mark

* changing the import

* removing the setting ID from imports

* adding code in order to update when the hover updates

* adding methods to service

* adding code in order to dispose the accessible hover view

* fixing bug

* polishing the code

* checking that action not supported for the early return

* using disposable store instead

* using the appropriate string

* polishing the code

* using instead the type help and the resolved keybindings

* hiding also on the `onDidBlurEditorWidget` firing

* Revert "using instead the type help and the resolved keybindings"

This reverts commit 1f450dd535c6c71b37a15e20bf48434e11b7badf.

* use hover accessible view, provide custom help

* Revert "Revert "using instead the type help and the resolved keybindings""

This reverts commit 12f0cf6143715b3972e15270b4eb113edbdfbf88.

* add HoverAccessibilityHelp, BaseHoverAccessibleViewProvider

* polishing the code

* polishing the code

* provide content at a specific index from the hover accessibility help provider

* introducing method _initializeOptions

* using readonly where possible

* using public everywhere

* using a getter for the actions

---------

Co-authored-by: meganrogge <megan.rogge@microsoft.com>

RomanNikitenko committed:
Rebase against the upstream 7f55a08120d3586a4fbe41e9d30188e809f74cbb
vscode-upstream-sha1: 7f55a08120d3586a4fbe41e9d30188e809f74cbb
rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

RomanNikitenko committed:
Merge branch 'main' into nodejs-20

alexr00 committed:
Add more info to comment range error (#213578)

Part of #213243

dependabot[bot] committed:
Bump postcss from 7.0.39 to 8.4.38 (#209857)

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.39 to 8.4.38.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.39...8.4.38)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SimonSiefke committed:
fix: dispose template data disposables in source column renderer (#202618)

aiday-mar committed:
fixes regression: adding a `isCheapToTokenize` check before tokenizing the line (#213674)

adding a check on how it is cheap to tokenize the lines

meganrogge committed:
fix windows terminal hint not showing up (#213695)

fix #213693

lramos15 committed:
Missing server node module (#213712)

daviddossett committed:
Fix hover padding regression (#213700)

daviddossett committed:
Fix textPreformatForeground for HC themes (#213698)

mjbvz committed:
Fix js/ts cross code block intellisense (#213726)

We need to register support for the backing copilot scheme too, not just the panel scheme in core

roblourens committed:
Fix vscode-api-tests in stable (#213733)

connor4312 committed:
debug: bump js-debug for 1.90 (#213735)

joyceerhl committed:
fix: render ranges for attached selections in chat (#213741)

meganrogge committed:
fix accessible view kb quickpick not showing (#213745)

joyceerhl committed:
fix: set aria label for chat context attachments (#213747)

justschen committed:
web issue reporter styles (#213751)

remove mac from tag

justschen committed:
defaults web issue report false and disable in Firefox (#213752)

* turn off in release and in ff

* update

shoham-utila committed:
remove global enablement message (#213128)

bpasero committed:
editors - clarify `window.restoreWindows` setting further (#213771)

bpasero committed:
`Ctrl + I` does not work when chat input field has focus (fix #213637) (#213776)

* `Ctrl + I` does not work when chat input field has focus (fix #213637)

* simplify

bpasero committed:
aux window - support Firefox (#213769)

* aux window - support Firefox

* add more

* only use workaround for FF

* enable issue repoter again

* .

Tyriar committed:
Merge branch 'main' into tyriar/pwsh5

Tyriar committed:
Merge pull request #211993 from microsoft/tyriar/pwsh5

Disable sending via CompletionCompleters for pwsh5

sandy081 committed:
fix #212208 (#213779)

Tyriar committed:
Make rest of terminal initial hint italic

Part of #213647

Tyriar committed:
Align colors of terminal hint with editor

Tyriar committed:
Use pointer cursor for whole terminal initial hint

Fixes microsoft/vscode-copilot#5963

Tyriar committed:
Merge pull request #213783 from microsoft/tyriar/213647

Align terminal initial hint styles with the editor

jrieken committed:
unwrap `LanguageModelError` when invoking chat participant (#213784)

re https://github.com/microsoft/vscode/issues/213710

jrieken committed:
fix https://github.com/microsoft/vscode-copilot/issues/5814 (#213787)

jrieken committed:
fixes https://github.com/microsoft/vscode-copilot/issues/5908 (#213788)

jrieken committed:
fixes  https://github.com/microsoft/vscode/issues/213644 (#213789)

jrieken committed:
fix https://github.com/microsoft/vscode/issues/213614 (#213791)

lszomoru committed:
SCM - all calls from extHostSCM to mainThreadSCM should use a sequencer to ensure that the source control object is created (#213792)

benibenj committed:
Fix multi selected tab close behavior (#213794)

fix #213620

sandy081 committed:
fix #196351 (#213796)

alexr00 committed:
Cannot read properties of undefined (reading 'layout') (#213797)

Fixes #212235

sandy081 committed:
fix #211965 (#213801)

Yesterday17 committed:
fix: remove temp dir if extension is installed by another source (#213379)

bpasero committed:
closing out inline chat while speech is going does not stop audio (fix #213709) (#213802)

bpasero committed:
No text to speech when a change is applied (fix #213667) (#213804)

alexr00 committed:
Regression: Unsaved comment reply lost when changing editors (#213805)

Fixes #213144

bpasero committed:
Voice dictation shortcut (fix #206800) (#213806)

Tyriar committed:
Merge branch 'main' into tyriar/copilot-5963

Tyriar committed:
Merge pull request #213785 from microsoft/tyriar/copilot-5963

Use pointer cursor for whole terminal initial hint

Tyriar committed:
Enable initialHint when not stable for now

Part of #213647

Tyriar committed:
Merge pull request #213815 from microsoft/tyriar/213647_2

Enable initialHint when not stable for now

jrieken committed:
fix leaking listener (#213816)

re https://github.com/microsoft/vscode/issues/213141

rzhao271 committed:
fix: setting labels appearing in widgets (#213826)

connor4312 committed:
Connor4312/issue211378 (#213829)

cli: fix inverted condition not showing warning correctly

Fixes #211378

jrieken committed:
doc updates & fixes for chat/lm API (#213777)

* workaround @link not working

* fixes https://github.com/microsoft/vscode/issues/213630

TylerLeonhardt committed:
Retain previous quick pick navigation behavior with Home and End (#213832)

Fixes https://github.com/microsoft/vscode/issues/213739

rebornix committed:
Fix #213525. Move nb chat variable to service (#213835)

joyceerhl committed:
fix: hide Attach Context for quick chat (#213841)

Yoyokrazy committed:
Hook up notebook format code action to format notebook command (#213847)

hook format codeaction to format notebook command

mjbvz committed:
Fix references for copilot backing docs again (#213848)

Fixes https://github.com/microsoft/vscode-copilot-release/issues/1055

roblourens committed:
Show proper warning for outdated Copilot Chat in the next build (#213852)

alexdima committed:
Ensure inline chat is started with a good initial selection (#213860)

roblourens committed:
'request' may be undefined when handling an error while sending a chat request (#213869)

Part of microsoft/vscode-copilot#5942

joyceerhl committed:
fix: ensure chat view is revealed when attaching context (#213871)

RomanNikitenko committed:
Rebase against the upstream bc15ede705b9168324a3440b7c80c80644743306
vscode-upstream-sha1: bc15ede705b9168324a3440b7c80c80644743306
rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

aiday-mar committed:
Adding back the CSS into the hover code (#213831)

* adding back the appropiate css

* removing the padding bottom maybe not needed?

roblourens committed:
Fix sendChatRequest reference (#213870)

aiday-mar committed:
Changing the wording in the accessible view/help (#213681)

changing the wording of the hover widget

roblourens committed:
Add more notes to selectChatModels (#213867)

aiday-mar committed:
Reusing NLS label of markdown verbose hover in accessible view (#213680)

* reusing nls label

* polishing

jrieken committed:
fixes https://github.com/microsoft/vscode/issues/213716 (#213828)

* fixes https://github.com/microsoft/vscode/issues/213716

* fix tests

bpasero committed:
voice - make sure speech context counters do not get out of sync (#213637) (#213898)

hediet committed:
Fixes #213703 (#213899)

Revert "Fixes #209417 (#213549)"

This reverts commit e9a5733f32725b5f26972cc6cfb9e4057ad2dc11.

jrieken committed:
improve zone revealing when showing above line 1 (#213904)

fixes https://github.com/microsoft/vscode/issues/213822

SimonSiefke committed:
fix wrong colors when editor findMatchForeground is not defined (#213686)

* fix css

Tyriar committed:
Workaround low underscore in fonts by adjusting line height

Part of #211933

Tyriar committed:
Check minor bash version before using @P

Fixes #212180

Tyriar committed:
Fix v4 major check

alexr00 committed:
Only fire collapsible change event if actually changed. (#213910)

jrieken committed:
On request-removal undo till the point at which the request was made (#213902)

* On request-removal undo till the point at which the request was made

fixes https://github.com/microsoft/vscode-copilot/issues/5736

* fixed EOL sequence which (hopefully) fixes tests

* give up, don't use new lines...

Tyriar committed:
Merge pull request #213908 from microsoft/tyriar/212180

Check minor bash version before using @P

chrmarti committed:
Use start position when going reverse (microsoft/vscode-copilot#5946)

Tyriar committed:
Fix test for new lineheight value

Tyriar committed:
Merge pull request #213905 from microsoft/tyriar/211933

Workaround low underscore in fonts by adjusting line height

jrieken committed:
Revive serialized chat data when restoring session (#213915)

fixes https://github.com/microsoft/vscode/issues/213914

roblourens committed:
Explain chat participant name requirement more (#213924)

Another note for #210385

meganrogge committed:
add setting description for general signal delays (#213927)

meganrogge committed:
fallback to widget hover content if no markdown hover content found (#213932)

fix #213931

justschen committed:
move web reporter setting and check version (#213934)

* move registry and check at run time

* remove whitespace

meganrogge committed:
make signalOptions delays experimental  (#213938)

fix #213623

jrieken committed:
Fix issue with "reun without" while request is pending (#213926)

- send out `ChatRequestRemovalReason`
- make inline chat re-enter `SHOW_REQUEST` when resending an active request
- the `SHOW_REQUEST` state should be prepared to come after the request has started
- more cancellation checking with `asProgressiveEdit`
- add tests

https://github.com/microsoft/vscode-copilot/issues/5997

bhavyaus committed:
Add null check for session in accountsEntitlements (#213933)

meganrogge committed:
ensure chat response isn't too verbose (#213939)

fix #212788

bhavyaus committed:
Update distro (#213955)

meganrogge committed:
add signalOptions description (#213957)

fix #213670

mjbvz committed:
Don't use shell for spawning node (#213960)

Fixes #204678

lramos15 committed:
Bump versions (#213961)

connor4312 committed:
testing: fix title for show/hide inline coverage (#213946)

Fixes #213830

connor4312 committed:
testing: allow test runs to run tests outside of the original scope (#213945)

Fixes #213182

connor4312 committed:
forwarding: make https work for port forwarding (#213943)

Closes https://github.com/microsoft/vscode/issues/201465

RomanNikitenko committed:
Rebase against the upstream 54dd0ecc653b89e6032369c6f4b1808bc8e37ec5
vscode-upstream-sha1: 54dd0ecc653b89e6032369c6f4b1808bc8e37ec5

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

jrieken committed:
fix https://github.com/microsoft/vscode/issues/213970 (#213977)

lszomoru committed:
Git - add error handling for traversing workspace folders during repository discovery (#213975)

jrieken committed:
collect metadata about workspace stats collection (#213982)

jrieken committed:
Fix empty diff editor, leakage, and duplicated labels (#213993)

Store/dispose a reference to the RHS model of the compare editor instead of the model, make sure diff editors are released, fix repeated labels

fixes https://github.com/microsoft/vscode-copilot/issues/5885
fixes https://github.com/microsoft/vscode/issues/213992

sandy081 committed:
proper fix #213971 (#213995)

jrieken committed:
refine "workspace.stats.metadata" (#214000)

aiday-mar committed:
Adding the padding border back into `hover.css` file (#214001)

adding the padding-border back

alexr00 committed:
Fix comment not expanding (#213985)

Fixes #213974

alexr00 committed:
Update grammars (#214005)

jrieken committed:
chore - tweak state validation logic to play nicer with our test integration (#214006)

sandy081 committed:
fix #213837 (#214007)

jrieken committed:
chore - remove unused/dead code (#214009)

* remove dead `IInlineChatResponse`

* remove `IInlineChatSession`

jrieken committed:
add unit test to confirm https://github.com/microsoft/vscode-copilot/issues/5442 (#214012)

jrieken committed:
Fix layout issues in inline chat (#214013)

* clamp followup height to three lines

fixes https://github.com/microsoft/vscode-copilot/issues/5522

* follow up question should not define the widget of the chat input part

bpasero committed:
Allow to pass language mode for `workbench.action.editor.changeLanguageMode` action (fix #213688) (#213912)

bpasero committed:
perf - fix counting of major GCs (#214016)

connor4312 committed:
ports: fix protocol switcher not working locally (#213951)

Fixes #213944

It previously always targeted the USER_REMOTE settings when changing the
protocol, but these did not exist locally.

gjsjohnmurray committed:
Don't merge a view with a titleDescription (fix #166000) (#214010)

lszomoru committed:
SCM - adopt observables for commitTemplate (#214021)

amunger committed:
instantiate status bar contribs with services scoped to the window (#211997)

jrieken committed:
fix a few leaking listeners (#214024)

jrieken committed:
make sure inline chat content widget doesn't show empty message bar (#214025)

Tyriar committed:
Disable initialHint after inline chat is triggered

Fixes #213820

Tyriar committed:
Reset initial hint state when config changes

roblourens committed:
Fix search interface name (#214029)

Fix interface name

Tyriar committed:
Reset state in right spot

Tyriar committed:
Use the default agent name in terminal initial hint

Fixes microsoft/vscode-copilot#5969

tisilent committed:
Add showIncludesExcludes in IFindInFilesArgs (#212347)

* Add showIncludesExcludes in IFindInFilesArgs

* :lipstick:

rzhao271 committed:
fix: some setting links broken in zh-cn (#214035)

lszomoru committed:
SCM - fix listener leak (#214037)

Tyriar committed:
Merge pull request #214032 from microsoft/tyriar/copilot-5969

Use the default agent name in terminal initial hint

andreamah committed:
clean up aiTextSearchProvider (#214039)

roblourens committed:
Stream chinese text in chat character-by-character (#214046)

Fix #213287

bpasero committed:
Panel visibility is not persisted/restored on vscode.dev after restart (fix #213885) (#214055)

* Panel visibility is not persisted/restored on vscode.dev after restart (fix #213885)

* :lipstick:

SimonSiefke committed:
feature: replace electron File.path with electron webUtils (#213031)

* add getPathForFIle function

* use electron webutils

* use webutils

* use getPathForFile

* add test

* fix import

* add web utils service

* register electron webutils service

* register contribution

* import contribution

* register web contribution

* feature: use webUtils service to not directly depend on electron global

* use webutils service for terminal

* update imports

* tsc

* update imports

* add servicebrand

* tsc

* tsc

* try different approach

* add class and subclass

* register contribution with subclass

* rename function

* remove web utils service

* register getPathForFile function

* fix import

* remove webutils service

* remove import

* remove import

* fix imports

* fix import

* fix overwriting view

* register method

* fix import

* register contribution

* don't overwrite electron service with browser service

* remove log

* remove log

* tsc

* Discard changes to src/vs/workbench/contrib/files/browser/fileActions.contribution.ts

* Discard changes to src/vs/workbench/contrib/files/browser/fileActions.ts

* Discard changes to src/vs/workbench/contrib/files/electron-sandbox/fileActions.contribution.ts

* Discard changes to src/vs/workbench/contrib/terminal/browser/terminal.contribution.ts

* Discard changes to src/vs/workbench/contrib/terminal/browser/terminal.ts

* Discard changes to src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

* Discard changes to src/vs/workbench/contrib/terminal/browser/terminalInstanceService.ts

* Discard changes to src/vs/workbench/contrib/terminal/browser/terminalTabbedView.ts

* Discard changes to src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts

* Discard changes to src/vs/workbench/contrib/terminal/browser/terminalView.ts

* Discard changes to src/vs/workbench/contrib/terminal/electron-sandbox/terminal.contribution.ts

* Discard changes to src/vs/workbench/contrib/terminal/electron-sandbox/terminalInstanceService.ts

* Discard changes to src/vs/workbench/contrib/terminal/electron-sandbox/terminalView.ts

* Discard changes to src/vs/workbench/contrib/terminal/test/browser/terminalInstance.test.ts

* add function to interface

* add function

* use host service

* use host service in terminal tabs list

* use host service in terminal instance

* remove unused code

* fix imports

* replace remaining usages of file path

* tsc

* add semicolon

* cleanup

---------

Co-authored-by: Benjamin Pasero <benjamin.pasero@microsoft.com>

roblourens committed:
Show chat agent hover in chat input editor (#214105)

* Show chat agent hover in chat input editor

* Clean up

* Comment

RomanNikitenko committed:
Rebase against the upstream 403294d92b4fbcdad61bb74635a8e5e145151aaa
vscode-upstream-sha1: 403294d92b4fbcdad61bb74635a8e5e145151aaa

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

alexr00 committed:
Remove padding from inline codespans in comment widget (#214123)

Inline codespans in comment widget don't need padding
Fixes #214038

RomanNikitenko committed:
Merge branch 'main' into nodejs-20

jrieken committed:
chore - remove dead CSS variable usage (#214125)

alexr00 committed:
Dispose scoped instantiation service in comments (#213808)

RomanNikitenko committed:
Fix build

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

bpasero committed:
Show progress indication when saving an editor takes long (fix #193230) (#214128)

bpasero committed:
status bar - make `loading` the default (#214135)

This is a more generic progress indicator than `syncing` and looks better.

alexr00 committed:
Fix a listener in tunnel service that's GC'ed without having been disposed (#214138)

aiday-mar committed:
Refactoring the stickyScrollController `setState` method (#213917)

* adding changes

* fixing compilation errors, using undefined as the fallback type not null

bpasero committed:
debt - :up: `jschardet@3.1.2` (#214136)

TylerLeonhardt committed:
Handle when `item` is undefined (#214107)

Fixes https://github.com/microsoft/vscode/issues/213363

joaomoreno committed:
upload sourcemaps to CDN instead of ticino (#214141)

jrieken committed:
configure `esModuleInterop: true`, change some star-imports

amunger committed:
remove important! on style (#214044)

jrieken committed:
change all assert-star-imports

jrieken committed:
adding missing bundle.js compile output

Tyriar committed:
Merge pull request #214028 from microsoft/tyriar/213820

Disable initialHint after inline chat is triggered

andreamah committed:
dispose child instantiation service in search (#214145)

Tyriar committed:
xterm@5.6.0-beta.18

Fixes #214142

Tyriar committed:
Merge branch 'main' into tyriar/214142

meganrogge committed:
update default value for setting to `.experimental` (#214149)

fix #213968

amunger committed:
get the serializer from the notebook service each time (#214036)

* get the serializer from the notebook service each time

* add logservice parameter

* fix unit test

Tyriar committed:
Merge pull request #214150 from microsoft/tyriar/214142

xterm@5.6.0-beta.18

jrieken committed:
tweak SWC options to allow module interop

bhavyaus committed:
[Engineering] Update locker.yml to skip locking closed backlog candidate issues (#214160)

Update locker.yml to ignore closed backlog candidate issues

amunger committed:
switch default interactive.execute keybinding (#214161)

switch default keybinding

roblourens committed:
Use correct offset when removing instruction breakpoints (#214162)

Fix #212496

meganrogge committed:
deconstruct `signalOptions` config object, migrate settings (#214163)

fix #213618

joyceerhl committed:
refactor: move Add Context... actions to core (#214167)

connor4312 committed:
debug: fix allowing logpoints to be set and triggering bps (#214168)

Fixes #213216

mjbvz committed:
Pick up TS 5.5 for builtin ts extension (#214170)

For #214169

mjbvz committed:
Pick up latest TS for building VS Code (#214171)

meganrogge committed:
do not show terminal hint for reconnected terminals (#214174)

fix #212278

Tyriar committed:
xterm@5.6.0-beta.19

Fixes #213174
Fixes #213304

roblourens committed:
Count codeblocks correctly in basicRenderElement (#214179)

Not trying to fix progressive rendering here
Better fix for #214071

Tyriar committed:
Merge pull request #214181 from microsoft/tyriar/213174

xterm@5.6.0-beta.19

connor4312 committed:
testing: fix testController.items.delete is not removing the item from the tree UI (#214182)

Fixes #213316

joyceerhl committed:
chore: remove empty editor chat hint setting migration

joyceerhl committed:
feat: right click to hide empty text editor hint

connor4312 committed:
debug: fix hover disappearing 'too quickly' (#214186)

What was actually happening is that in the process of hovering into the
debug hover, another position was briefly hovered. No data was available
for this position, so the hover was hidden.

This PR implements "safe triangle" logic to avoid changing the hover position if the mouse moves in the safe triangle created by a displayed hover widget.

Fixes #212924

andreamah committed:
not specifying `maxResults` on `findTextInFiles`->`textSearchProvider` will use `MAX_INT` for `maxResults`
Fixes #214189

andreamah committed:
more instances of max safe integer

jrieken committed:
Merge branch 'main' into joh/intense-heron

jrieken committed:
print loader error

sandy081 committed:
improvements to profiles editor (#214198)

* feedback

* use list

* reduce size

* fix checkbox and selectbox

* preview profile

* rename

* disable while creating and previewing

* improvements

* refactor

* fix

jrieken committed:
remove graceful-fs error message

lszomoru committed:
SCM - 💄 use observables for count and statusBarCommands (#214202)

bpasero committed:
esm - remove `graceful-fs` (#214164)

* esm - remove `graceful-fs`

This module was used to patch `fs`, which is not supported in ESM. Besides, it makes us behave different from standard node.js that has meanwhile evolved from the time where `graceful-fs` was created.

* update comment

* update comment

* use more `fs.promises`

* :lipstick:

bpasero committed:
Merge branch 'main' into joh/intense-heron

aiday-mar committed:
Store `minRebuildRromLine` field so sticky scroll is ultimately correctly rerendered (#213916)

* adding changes

* on this branch we have all the other changes

* using the next rebuild from line variable

* adding changes

* polishing the code

jrieken committed:
Merge pull request #214144 from microsoft/joh/intense-heron

Use `esModuleInterop` flag, adopt star-imports

lszomoru committed:
SCM - 💄 remove the use of a sequencer (#213997)

* Revert "SCM - all calls from extHostSCM to mainThreadSCM should use a sequencer to ensure that the source control object is created (#213792)"

This reverts commit 24dd222ff2b0f5fb70ef1daa1f00783bbf7669ae.

* Initial implementation

* Pull request feedback

* Revert unnecessary change

* Use map of barriers instead

* 💄 rename

aiday-mar committed:
Refactoring the `contentHoverController.ts` file (#214214)

refactoring the content hover controller file

alexr00 committed:
Comment text cut off in comments panel (#214212)

Fixes #212956

lszomoru committed:
SCM - fix input theming (#214225)

lszomoru committed:
SCM - fix timing issue with setting context keys (#214229)

lszomoru committed:
Engineering - update winget configuration file (#214228)

* Engineering - update winget configuration file

* Fix the version number

alexdima committed:
Fixes #207423 (#214232)

alexdima committed:
Fixes #207433 (#214233)

hediet committed:
improves obsCodeEditor helper (#214215)

* improves obsCodeEditor helper

hediet committed:
Fixes #214049

hediet committed:
Merge pull request #214241 from microsoft/hediet/b/sophisticated-canid

Fixes #214049

andreamah committed:
Merge pull request #214190 from microsoft/andreamah/issue214189

should not use `max_int` as default for for `maxResults`

connor4312 committed:
serve-web: fix `SECRET_KEY_MINT_PATH` does not honour `server-base-path` (#214250)

Fixes #212369

alexr00 committed:
Add API proposal to reveal a comment thread (#214251)

Fixes #167253

amunger committed:
dont calculate element dimensions before styles are loaded (#214249)

* dont calculate element dimensions before styles are loaded

* remove unused import

sandy081 committed:
more profile editor improvements (#214255)

Tyriar committed:
Register some listeners

Part of #214234

Tyriar committed:
Remove CommandDetectionCapability.hasInput

Fixes #213694

Tyriar committed:
Merge pull request #214258 from microsoft/tyriar/214234

Register some listeners

Tyriar committed:
Fix typo in shell integration api docs

connor4312 committed:
testing: fix "Run Tests" explorer option does not discover tests before trying to run (#214261)

Fixes #212342

Tyriar committed:
Merge pull request #214259 from microsoft/tyriar/213694

Remove CommandDetectionCapability.hasInput

Tyriar committed:
Merge pull request #214263 from microsoft/tyriar/api_fix

Fix typo in shell integration api docs

alexdima committed:
Fixes #207457 (#214267)

alexdima committed:
Avoid leaking disposables (#214234) (#214268)

connor4312 committed:
testing: ensure run all runs everything, use consistent profile for re-runs (#214260)

* testing: ensure run all runs everything, use consistent profile for re-runs

Fixes #212628

* fixup

* fix compile

connor4312 committed:
testing: fix new test runs are not collapsible (#214271)

Fixes #212875

meganrogge committed:
dispose terminal hint decoration on config change (#214178)

meganrogge committed:
mute terminal hint color (#214173)

fix #213647

hediet committed:
Inline completions refactoring (#214272)

connor4312 committed:
testing: fix confusing line decorations when hovering over line numbers (#214275)

Fixes #212585

franciscacarneiro committed:
Fix #182308: Pressing PageUp in the search panel no longer causes the layout to change. (#213067)

Bug (previous behavior):
    1.Press Ctrl+Shift+F to open the search panel.
    2.Enter any text into the search input.
    3.Press PageUp/PageDown.
    4.The layout shifts to the right, hiding the line that indicates the current tab in the activity bar.

Solution:
When search results are displayed, pressing PageUp or PageDown now moves the cursor to the start or end of the search input, respectively. This is achieved by assigning specific functionality to these keys, overriding their default behavior of moving through the page.

Testing:
A smoke test was implemented to assess the fix. The test involves entering text into the search editor, pressing the PageUp and PageDown keys, and checking for any changes in the activity bar's layout.

connor4312 committed:
debug: support 'source' in group output events (#214278)

Fixes #212304

remcohaszing committed:
Replace `removeChild` with `remove` (#213465)

* Replace removeChild with remove

This replaces most uses of `parent.removeChild(child)` with
`child.remove()`.

The two are almost equivalent. The only difference is that
`parent.removeChild(child)` throws if the given node is not a child of
the parent, whereas `child.remove()` never throws. There is no noticable
performance difference. The only reason to use `removeChild` is to
support Internet Explorer, but that’s no longer supported by Monaco
editor.

* Fix broken CSP hash

The script content changed, so the sha256 hash changed too.

* Update src/vs/editor/browser/config/charWidthReader.ts

Co-authored-by: Logan Ramos <lramos15@gmail.com>

---------

Co-authored-by: Logan Ramos <lramos15@gmail.com>

connor4312 committed:
eng: dispose child instantiation service instances in debug and testing (#214279)

Refs #212879

alexdima committed:
Fixes #207445 (#214266)

* Fixes #207445

* Fix problem with disposing

rebornix committed:
Re #211878. Fix unsafe type assertions for notebooks (#214280)

rebornix committed:
Re #212879. track child instantiation service for notebooks (#214281)

alexdima committed:
Reduce casts (#211878) (#214282)

alexdima committed:
Fixes #207802 (#214284)

joyceerhl committed:
refactor: remove `includeSymbols` from quick access (#214288)

connor4312 committed:
userActivity: maintain user activity while there's progress in the window (#214264)

Fixes #211064

sandy081 committed:
adopt to new vsce-sign (#214300)

jrieken committed:
fix leaking listeners, track dispoable workbench contributions (#214301)

* fix leaking listener

https://github.com/microsoft/vscode/issues/214234

* track disposable workbench contribution and dispose them on shutdown fyi @bpasero

https://github.com/microsoft/vscode/issues/214234

* fix leaking listener

https://github.com/microsoft/vscode/issues/214234

sandy081 committed:
update-distro (#214302)

jrieken committed:
fix leaking listener (#214305)

* polish GC listener warning workflow

re https://github.com/microsoft/vscode/issues/214234

* fix leaking listener

https://github.com/microsoft/vscode/issues/214234

hediet committed:
Fixes CI failure (#214307)

* Fixes CI failure

* Fixes CI

aiday-mar committed:
Avoid auto-reindentation when pasting text with only whitespace characters (#213921)

* adding draft code for checking for empty strings

* polishing code

hediet committed:
Enables CodeEditorWidget test suite again (#214309)

yutotnh committed:
Add the ability to specify a list of candidate encodings when guessing encoding (#36951) (#208550)

* Allow to configure a list of encodings to use when guessing #36951

* Bump up the jschardet version into 3.1.2 #36951

* missing merge

* some polish

* renames

* some polish

* some polish

* cleanup

---------

Co-authored-by: Benjamin Pasero <benjamin.pasero@microsoft.com>

aiday-mar committed:
Extract code from `ContentHoverController` constructor into separate methods (#214316)

extracting code from ContentHoverController into separate methods

bpasero committed:
Make sure event subscriptions are stored and disposed (#214234) (#214318)

alexdima committed:
Better track disposables (#214234) (#214322)

RomanNikitenko committed:
Rebase against the upstream 3024b310316d52c25895b3ecf0a2d99e65d4e6fb
vscode-upstream-sha1: 3024b310316d52c25895b3ecf0a2d99e65d4e6fb

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

RomanNikitenko committed:
Merge branch 'main' into nodejs-20

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

RomanNikitenko committed:
Do not verify signature at an extension installation

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

RomanNikitenko committed:
Enable extension signature verification

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

RomanNikitenko committed:
Merge branch 'main' into nodejs-20
